src/Jackett.Common/Indexers/Rarbg.cs

```diff
@@ -295,26 +192,72 @@ private async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuery query, in
             return releases;
         }
 
-        public override async Task<byte[]> Download(Uri link)
+        private string BuildSearchUrl(TorznabQuery query)
         {
-            // build download link from info redirect link
-            var slink = link.ToString();
-            var response = await RequestStringWithCookies(slink);
-            if (!response.IsRedirect && response.Content.Contains("Invalid token."))
+            var searchString = query.GetQueryString();
+            var qc = new NameValueCollection
+            {
+                { "token", _token },
+                { "format", "json_extended" },
+                { "app_id", _appId },
+                { "limit", "100" },
+                { "ranked", "0" },
+                { "sort", _sort }
+            };
+
+            if (query.ImdbID != null)
+            {
+                qc.Add("mode", "search");
+                qc.Add("search_imdb", query.ImdbID);
+            }
+            else if (query.RageID != null)
+            {
+                qc.Add("mode", "search");
+                qc.Add("search_tvrage", query.RageID.ToString());
+            }
+            /*else if (query.TvdbID != null)
+            {
+                queryCollection.Add("mode", "search");
+                queryCollection.Add("search_tvdb", query.TvdbID);
+            }*/
+            else if (!string.IsNullOrWhiteSpace(searchString))
+            {
+                searchString = searchString.Replace("'", ""); // ignore ' (e.g. search for america's Next Top Model)
+                qc.Add("mode", "search");
+                qc.Add("search_string", searchString);
+            }
+            else
             {
-                // get new token
-                token = null;
-                await CheckToken();
-                slink += "&token=" + token;
-                response = await RequestStringWithCookies(slink);
+                qc.Add("mode", "list");
+                qc.Remove("sort");
             }
-            if (!response.IsRedirect)
-                throw new Exception("Downlaod Failed, expected redirect");
 
-            var targeturi = new Uri(response.RedirectingTo);
-            var id = targeturi.Segments.Last();
-            var dluri = new Uri(targeturi, "/download.php?id=" + id + "&f=jackett.torrent");
-            return await base.Download(dluri, RequestType.GET);
+            var querycats = MapTorznabCapsToTrackers(query);
+            if (querycats.Count == 0)
+                querycats = GetAllTrackerCategories(); // default to all, without specifing it some categories are missing (e.g. games), see #4146
+            var cats = string.Join(";", querycats);
+            qc.Add("category", cats);
+
+            return ApiEndpoint + "?" + qc.GetQueryString();
+        }
+
+        private async Task RenewalToken(bool force = false)
```

Comment:
Rename to match Naming convention.
```suggestion
        private async Task RenewalTokenAsync(bool force = false)
```